reporter-sqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,203 @@
+
+  /** close is typically called from a different thread */
+  private AtomicBoolean closeCalled = new AtomicBoolean(false);
+
+
+  @Override public CheckResult check() {
+    if (performCheck()) {
+      try {
+        get().getQueueAttributes(queueUrl(), Collections.singletonList("QueueArn"));
+        return CheckResult.OK;
+      } catch (Throwable e) {
+        return CheckResult.failed(new Exception(e));
+      }
+    }
+
+    return CheckResult.OK;
+  }
+
+  @Override protected AmazonSQSBufferedAsyncClient compute() {
+    AWSCredentialsProvider creds = (credentialsProvider() != null) ?
+        credentialsProvider() : new DefaultAWSCredentialsProviderChain();
+
+    AmazonSQSAsyncClient client;
+    if (clientConfiguration() != null && dispatchExecutor() != null) {
+      client = new AmazonSQSAsyncClient(creds, clientConfiguration(), dispatchExecutor());
+    } else if (clientConfiguration() != null && dispatchExecutor() == null) {
+      client = new AmazonSQSAsyncClient(creds, clientConfiguration());
+    } else if (clientConfiguration() == null && dispatchExecutor() != null) {
+      client = new AmazonSQSAsyncClient(creds, dispatchExecutor());
+    } else {
+      client = new AmazonSQSAsyncClient(creds);
+    }
+
+    return new AmazonSQSBufferedAsyncClient(client);
+  }
+
+  @Override public int messageSizeInBytes(List<byte[]> encodedSpans) {
+    return encoding().listSizeInBytes(encodedSpans);
+  }
+
+  @Override public void sendSpans(List<byte[]> list, Callback callback) {
+    if (closeCalled.get()) throw new IllegalStateException("closed");
+
+    checkNotNull(list, "list of encoded spans must not be null");
+
+    byte[] encodedSpans = BytesMessageEncoder.forEncoding(encoding()).encode(list);
+
+    get().sendMessageAsync(new SendMessageRequest(queueUrl(), "z")
+        .addMessageAttributesEntry("z", new MessageAttributeValue()
+            .withDataType("Binary." + encoding().name())
+            .withBinaryValue(ByteBuffer.wrap(encodedSpans))), new AsyncHandlerAdapter(callback));
+  }
+
+  @Override public void close() {
+    if (!closeCalled.getAndSet(true)) {
+      AmazonSQSBufferedAsyncClient maybeNull = maybeNull();
+      if (maybeNull != null) maybeNull.shutdown();
+    }
+  }
+
+  private static ThreadFactory threadFactory(final String name, final boolean daemon) {
+    return (Runnable r) -> {
+      Thread result = new Thread(r, name);
+          result.setDaemon(daemon);
+          return result;
+    };
+  }
+
+  private static class AsyncHandlerAdapter implements AsyncHandler<SendMessageRequest, SendMessageResult> {
+
+    private final Callback callback;
+
+    AsyncHandlerAdapter(Callback callback) {
+      this.callback = callback;
+    }
+
+    @Override public void onError(Exception e) {
+      callback.onError(e);
+    }
+
+    @Override
+    public void onSuccess(SendMessageRequest request, SendMessageResult sendMessageResult) {
+      callback.onComplete();
+    }
+  }
+}
```

Comment:
hide the AwsBufferedSqsSender constructor by defining a package private one here. this will make the type effectively package final